src/main/webapp/app/exam/participate/exam-participation.service.ts

```diff
@@ -129,9 +130,10 @@ export class ExamParticipationService {
                     if (!!participation.submissions) {
                         for (const submission of participation.submissions) {
                             delete submission.participation;
-                            if (!!submission.result) {
-                                delete submission.result.participation;
-                                delete submission.result.submission;
+                            if (!!getLatestSubmissionResult(submission)) {
```

Comment:
Quick question: isn't `!!getLatestSubmissionResult(submission)` the same as `!!getLatestSubmissionResult(submission)` (so that we can simply omit the `!!`)?

Reply:
In theory `!!` forces the expression to evaluate to a boolean, so that is likely the reason it is used here.

Reply:
As @FrankeLukas stated correctly, we enforce an evaluation as a boolean :)